chapter2.adoc

```diff
@@ -44,41 +44,29 @@ When `mdbgen` is set to 0, the <<dbops, debug operations>> are disallowed and th
 ==== Supervisor Domain Debug Control
 The Smsdedbg extension cite:[smmtt] introduces `sdedbgalw` field (bit 7) in CSR <<Sdseccsr,msdcfg>>, to control the debuggability of supervisor domains. When `mdbgen`=0, the `sdedbgalw` field determines both the debug allowed privilege modes and the <<dbgaccpriv, debug access privilege>>, as illustrated in <<dbgpriv>>. 
 
+[[dbgaccpriv]]
+==== Debug Access Privilege
+The *debug access privilege* is defined as the privilege level granted to the external debugger when performing state accesses via the hart, such as Abstract Commands and Program Buffer execution. Memory and register accesses from Debug Mode carry *debug access privilege* instead of always with M-mode, while the privilege is determined by <<mdbgctl, `mdbgen`>> and <<submdbgctl, `sdedbgalw`>>, as listed in <<dbgpriv>>. Illegal access will return an error and set `abstractcs.cmderr` to 3.
```

Comment:
I would replace the last two sentences with:
Where previously Debug Mode always had access to M-mode state, Debug Mode now accesses registers and memory using *debug access privilege*.  Attempts from Debug Mode to access state that requires privilege above *debug access privilege* will fail and set `abstractcs.cmderr` to 3.  The *debug access privilege* is derived as shown in [dbgpriv], and can be modified using `dcsr.dmprv` as described in [link to dmprv section].

Reply:
I would also move the `dmprv` definition section right below debug privilege table below.